image-diff

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+
+from machine import testvm
+
+
+def get_packages(machine):
+    if any(machine.image.startswith(p) for p in ['debian', 'ubuntu']):
+        pkgcmd = "dpkg-query -W -f='PACKAGE\t${source:Package}\t${version}\n'"
+    elif any(machine.image.startswith(p) for p in ['rhel', 'fedora', 'centos']):
+        # SOURCERPM is not great and the others don't work
+        pkgcmd = "rpm -qa --qf 'PACKAGE\t%{NAME}\t%{EVR}\n'"
+    else:
+        raise ValueError("Can only list packages on rpm/dpkg, not " + machine.image)
+
+    output = machine.execute(pkgcmd)
+
+    packages = {}
+    for line in output.splitlines():
+        if line.startswith('PACKAGE'):
```

Comment:
Seems redundant to have this PACKAGE prefix? These commands are meant to print exactly what you ask them to print, not anything extra. (But that's not a veto)

Reply:
I added this when I was dealing with `vm-run` and getting some extra output.  I could probably remove it now.

Reply:
This resulted in a *very* nice cleanup.  Thanks!